### ISSUE TITLE
Lower F1 Score

### ISSUE THREAD

**Issue by TonyPeng-2018** [WanhaoZhang/LogRAG#2]
Hello, Authors; I replicated your work following the README file. However, I got a pretty low F1 score on the example BGL dataset. How could I check where the issue is?

INFO:root:fixing windows size 1:  TP: 2, FP: 9, TN: 367, FN: 22 
INFO:root:fixing windows size 1: Acc: 0.9225, Precision: 0.1818, Recall: 0.0833, F1: 0.1143

INFO:root:fixing windows size 200, Tp: 2, Fp: 0, Tn: 0, Fn: 0
INFO:root:fixing windows size 200, Acc: 1.0000, Precision: 1.0000, Recall: 1.0000, F1: 1.0000

INFO:root:time windows size 1800 seconds, Tp: 2, Fp: 5, Tn: 116, Fn: 15
INFO:root:time windows size 1800 seconds, Acc: 0.8551, Precision: 0.2857, Recall: 0.1176, F1: 0.1667

If there is else I need to provide, please let me know; thanks.

**Comment by WanhaoZhang** [comment]
Hello, could you share your experiment parameters (including whether you used LogRAG-ChatGPT or LogRAG-Mistral)? If possible, please also provide the output from your training process. I can help you troubleshoot the issue.

**Comment by ethanthai2** [comment]
I have the exact same result as @TonyPeng-2018.

I use the LogRAG-ChatGPT

My parameters:

# deepsvdd parameters
normal_class: 0
is_pretrain: False
optimizer_name: adam
lr: 0.0001
n_epochs: 150
lr_milestones: [50]
batch_size: 40960
weight_decay: 0.0005
device: cuda:0
n_jobs_dataloader: 0

# rag parameters
threshold: 0.8
topk: 5
prompt: prompt5
persist_directory: ./output/ragdb-bgl

which is exactly the same as you put it in the config file.


My training result:

INFO:root:  Epoch 148/150        Time: 0.007     Loss: 0.02271469
INFO:root:  Epoch 149/150        Time: 0.007     Loss: 0.02271446
INFO:root:  Epoch 150/150        Time: 0.007     Loss: 0.02271425
INFO:root:Training time: 1.097
INFO:root:Finished training.
INFO:root:start testing....
INFO:root:Loading model from ./output/model.tar
Some weights of BertModel were not initialized from the model checkpoint at ./pretrained and are newly initialized: ['bert.pooler.dense.bias', 'bert.pooler.dense.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 400/400 [00:01<00:00, 301.55it/s]
INFO:root:Starting testing...
INFO:root:Testing time: 0.003
INFO:root:Test set AUC: 71.70%

**Comment by WanhaoZhang** [comment]
Hello,I believe that some changes may have been made to the ChatGPT-3.5 model compared to the version I used during my experiment, which could have led to deviations in the experimental results. I will try different versions of the GPT-3.5 model to investigate this further.
Additionally, I recommend also conducting experiments with the open-source model LogRAG-Mistral , as open-source models remain unchanged, ensuring more consistent and deterministic results.
Let me know if you need any further assistance.
